tests/unit/test_messages_emitter.py

```diff
@@ -1167,3 +1177,70 @@ def test_reporterror_command_error_no_stderr(get_initiated_emitter, stderr):
         call().show(sys.stderr, "test message", use_timestamp=False, end_line=True),
         call().stop(),
     ]
+
+
+# -- Tests for confirming a yes/no question with the user.
+
+def test_confirm_with_user_defaults_with_tty(
+    get_initiated_emitter,
+    emitter_mode: EmitterMode,
+    mock_input,
+    mock_isatty
+):
+    mock_input.return_value = ""
+    mock_isatty.return_value = True
+    emit = get_initiated_emitter(emitter_mode)
+
+    assert emit.confirm("prompt", default=True) is True
+    assert mock_input.mock_calls == [call("prompt [Y/n]: ")]
+    mock_input.reset_mock()
+
+    assert emit.confirm("prompt", default=False) is False
+    assert mock_input.mock_calls == [call("prompt [y/N]: ")]
+
+
+def test_confirm_with_user_defaults_without_tty(get_initiated_emitter, emitter_mode, mock_input, mock_isatty):
+    mock_isatty.return_value = False
+    emit = get_initiated_emitter(emitter_mode)
+
+    assert emit.confirm("prompt", default=True) is True
+    assert emit.confirm("prompt", default=False) is False
+
+    assert mock_input.mock_calls == []
+
+
+@pytest.mark.parametrize(
+    ("user_input", "expected"),
+    [
```

Comment:
How about some test cases with leading whitespace to exercise `strip()`?

Reply:
Done!